mods/mc_student/init.lua

```diff
@@ -69,132 +115,157 @@ local function show_report(player)
 	minetest.show_formspec(pname, "mc_student:report", table.concat(mc_student_report,""))
 	return true
 end
+-----------------------
+
+
+-----------------------
+--- MARKER ---
+local mc_student_marker = {
+	"formspec_version[5]",
+	"size[7,6.5]",
+	"position[0.3,0.5]",
+	"label[1.8,0.8;Add text to your marker]",
+	"button[0.7,5.2;2,0.8;back;Back]",
+	"button_exit[2.9,5.2;2,0.8;submit;Submit]",
+	"textarea[0.7,1.5;5.6,3.1;message;; ]"
+}
+
+	local function show_marker(player)
+	if mc_helpers.checkPrivs(player,priv_table) then
+		local pname = player:get_player_name()
+		minetest.show_formspec(pname, "mc_student:marker", table.concat(mc_student_marker,""))
+		return true
+	end
+end
+----------------------
+
+
+-------------------
+--- COORDINATES ---
+local selectedCoord = 0
+
+mc_student_coordinates = {
+	"formspec_version[6]",
+	"size[13,9.3]",
+	"label[5.4,0.4;Coordinates Stored]",
+	"button[9.3,7.6;3.3,0.6;record;Record]",
+	"button[9.3,8.5;1.6,0.6;delete;Delete]",
+	"button[11.2,8.5;1.4,0.6;go;Go]",
+	"button[7.2,8.5;1.7,0.6;clear;Clear All]",
+	"button[0.4,8.5;1.2,0.6;back;Back]",
+	"textarea[0.4,7.6;8.5,0.6;note;Add a note describing your current location;]",
+	"textlist[0.4,0.7;12.2,6.4;coordlist;;1;false]" 
+}
 
--- Define the Coordinates formspec
 local function show_coordinates(player)
-	local pname = player:get_player_name()
-	mc_student_coordinates = {
-		"formspec_version[5]",
-		"size[15,10]",
-		"label[6.3,0.5;Coordinates Stored]"
-	}
+	selectedCoord = 0
 
 	-- Get the stored coordinates for the player
+	local coordsList = {}
 	local pmeta = player:get_meta()
 	pdata = minetest.deserialize(pmeta:get_string("coordinates"))
+
 	if pdata == nil then
-		-- No coordinates stored, so return an empty list element
-		mc_student_coordinates[#mc_student_coordinates + 1] = "textlist[0.3,1;14.4,7.5;;No Coordinates Stored;1;false]"
+		table.insert(coordsList, "No Coordinates Stored")
 	else
-		mc_student_coordinates[#mc_student_coordinates + 1] = "textlist[0.3,1;14.4,7.5;coordlist;"
-		-- Some coordinates were found, so iterate the list
-		pxyz = pdata.coords
-		pnotes = pdata.notes
-		for i in pairs(pxyz) do
-			mc_student_coordinates[#mc_student_coordinates + 1] = pxyz[i] .. " " .. pnotes[i] .. ","
+		local prealms = pdata.realms
+		local pcoords = pdata.coords
+		local pnotes = pdata.notes
+
+		if pcoords then
+			for i in pairs(pcoords) do
+				local realm = Realm.GetRealm(prealms[i])
+				local pos = pcoords[i]
+				local utm = realm:WorldToUTMSpace(pos)
+				local latlong = realm:WorldToLatLongSpace(pos) 
+				local entry = realm.Name .. " "
+
+				if utm then
+					entry = entry .. math.floor(utm.x) .. "E " .. math.floor(utm.z) .. "N " .. math.floor(utm.y) .. "Z"
+				else
+					entry = entry .. "x=" .. math.floor(pos.x) .. " y=" .. math.floor(pos.y) .. " z=" .. math.floor(pos.z)
+				end
+
+				if latlong then
```

Comment:
Can we augment this so that if there are no UTM or latlong coordinates defined for the realm in map.conf, then it defaults to a format of `x=x , y=y, z=z`?

Reply:
Does this not currently do that (I haven't been able to test cases where the realm doesn't have utm/latlong due to mac issues)? Or are you just referring to adding the commas between the x y and z?

Reply:
No, currently the format is `xE zN yZ y.yyyy°N x.xxxx°W`.